### DISCUSSION TITLE
Questions about output of Ambiguity in groops

### DISCUSSION THREAD

**Discussion by FFFFFG-IGG** [groops-devs/groops#135] [Q&A]
Hello, Could someone offer some help about the descrbing information of groops's output ambiguity? It seems that thoese value is very small. I do not konw whether they are in cycles or in meters, and can not find clue about the description of that file(ambiguity.txt).

**Comment by tmayerguerr** [comment]
This is an internal file to test our block search algorithm. The first column contains the conditional variance. The other columns are the results for the intermediate steps for each solved block. They give the fractional part of the unresolved ambiguities (positive values) or the residuals for the resolved ambiguities (negative values) in cycles. There is no accounting of the integer number of cycles, since a rough estimate is already reduced from the observations during preprocessing.

For details see:
"On vectorial integer bootstrapping implementations in the estimation of satellite orbits and clocks based on small global networks", [doi.org/10.1016/j.asr.2021.09.023](https://doi.org/10.1016/j.asr.2021.09.023)
or the doctoral thesis of Sebastian [DOI 10.3217/978-3-85125-885-1](https://doi.org/10.3217/978-3-85125-885-1)

regards
Torsten

![ambiguities](https://user-images.githubusercontent.com/71824793/205483233-565a62ab-e110-4f9e-8d9a-02c3d174d351.png)
This figure shows an intermediate step (one column of the ambiguity file) of the block search algorithm.

**Reply by FFFFFG-IGG** [reply]
> This is an internal file to test our block search algorithm. The first column contains the conditional variance. The other columns are the results for the intermediate steps for each solved block. They give the fractional part of the unresolved ambiguities (positive values) or the residuals for the resolved ambiguities (negative values) in cycles. There is no accounting of the integer number of cycles, since a rough estimate is already reduced from the observations during preprocessing.
> 
> For details see: "On vectorial integer bootstrapping implementations in the estimation of satellite orbits and clocks based on small global networks", [doi.org/10.1016/j.asr.2021.09.023](https://doi.org/10.1016/j.asr.2021.09.023) or the doctoral thesis of Sebastian [DOI 10.3217/978-3-85125-885-1](https://doi.org/10.3217/978-3-85125-885-1)
> 
> regards Torsten
> 
> ![ambiguities](https://user-images.githubusercontent.com/71824793/205483233-565a62ab-e110-4f9e-8d9a-02c3d174d351.png) This figure shows an intermediate step (one column of the ambiguity file) of the block search algorithm.

Thanks, take an example of an ambiguity file derived from ppp as follow, my understand of what you say is that the first column contatins residuals of float ambiguities used for MLAMBDA(block search algorithm described in sebastian's doctor thesis). And in the second column, positive values represent residuals of fixed ambiguities while negative values represent fractional part of float ambiguites(not fixed). Is that right?
![image](https://user-images.githubusercontent.com/59271400/205591888-abd20e41-d0cc-480f-ba92-78ecd9718d33.png)
But there are still some questions about ambiguity file: the information that receiver-satellite pair which related to those ambiguites are unknown(I remember that groops previous version would output ambiguity with satellite infomation). So where can I get above infomation?
Thanks again,
Yongzhao Fan

**Comment by PierreS-alpha** [comment]
Dear GROOPS dev team,

I'm also interested in obtaining the ambiguities values out of a PPP processing. 
But if I understand  your answer correctly, there is no easy possibility to get the N or N·λ values out of this file, since this matrix contains only the residuals .
Are the ambiguity values possibly outputted by another module in another file?

Thanks a lot,
Pierre
